flink-core/src/main/java/org/apache/flink/core/fs/EntropyInjector.java

```diff
@@ -93,25 +93,18 @@ public static boolean isEntropyInjecting(FileSystem fs) {
 
     @Nullable
     private static EntropyInjectingFileSystem getEntropyFs(FileSystem fs) {
-        if (fs instanceof EntropyInjectingFileSystem) {
-            return (EntropyInjectingFileSystem) fs;
-        } else if (fs instanceof SafetyNetWrapperFileSystem) {
+        if (fs instanceof SafetyNetWrapperFileSystem) {
             FileSystem delegate = ((SafetyNetWrapperFileSystem) fs).getWrappedDelegate();
-            if (delegate instanceof EntropyInjectingFileSystem) {
-                return (EntropyInjectingFileSystem) delegate;
-            } else if (delegate instanceof PluginFileSystemFactory.ClassLoaderFixingFileSystem) {
-                FileSystem innerFs =
-                        ((PluginFileSystemFactory.ClassLoaderFixingFileSystem) delegate).getInner();
-                if (innerFs instanceof EntropyInjectingFileSystem) {
-                    return (EntropyInjectingFileSystem) innerFs;
-                }
-                return null;
-            } else {
-                return null;
-            }
-        } else {
-            return null;
+            return getEntropyFs(delegate);
         }
+
+        if (fs instanceof PluginFileSystemFactory.ClassLoaderFixingFileSystem) {
+            FileSystem innerFs =
+                    ((PluginFileSystemFactory.ClassLoaderFixingFileSystem) fs).getWrappedDelegate();
+            return getEntropyFs(innerFs);
+        }
```

Comment:
We can reduce both cases to check `if (fs instanceof WrappingProxy)` now and have it kind of future proof for future wrappers. I'd probably have the `EntropyInjectingFileSystem` check first (as it used to be) in case that some implementation will be both `EntropyInjectingFileSystem` and `WrappingProxy`.

Reply:
Thanks for detailed comment. just updated pr per suggestion and it seems cleaner implementation and future proof for Matryoshka doll like filesystem wrappers.